packages/lesmis/src/api/mutations/useLogin.js

```diff
@@ -27,5 +28,21 @@ export const useLogin = () => {
     },
   );
 
-  return query;
+  /**
+   * After the login fetch returns successfully, attempt to get the user and
+   * overwrite the login query state with the user query. This ensures that the user is
+   * not fully logged in until we have fetched a valid user and ensures any errors related to the
+   * getting a valid user are displayed at login.
+   */
+  const userQuery = useUser({ enabled: loginQuery.isSuccess });
+
+  if (loginQuery.isSuccess) {
+    return { ...loginQuery, ...userQuery };
+  }
+
+  /**
+   * If the login was not successful simply return the errors and fetching state related
+   * to the login query
+   */
+  return loginQuery;
```

Comment:
Could this go before the `useUser` as an early return? Or do you reckon the narrative reads better the way you've got it?

Reply:
Oh, is it because `useUser` needs to have `enabled: false` passed through so that future usage of it in other parts of the app will get some sort of disabled message?? Does the `enabled: false` persist to other calls until `enabled: true` is passed through?

Reply:
No no magic - as far as I'm aware the enabled prop won't get persisted or anything. The reason I didn't add the return before the useUser hook is because of one of the react [rules of hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) which is "Only Call Hooks at the Top Level".

Reply:
Aaah right! Of course, that makes sense. Thanks :-)